src/main/java/org/datadog/jenkins/plugins/datadog/util/git/RepositoryInfoCallback.java

```diff
@@ -24,25 +26,52 @@ public final class RepositoryInfoCallback implements RepositoryCallback<Reposito
     @Override
     public RepositoryInfo invoke(Repository repository, VirtualChannel channel) throws IOException, InterruptedException {
         try {
-            Ref head = repository.getRefDatabase().findRef("HEAD");
-            if(head == null) {
-                LOGGER.fine("Unable to build RepositoryInfo. HEAD is null.");
-                return RepositoryInfo.EMPTY_REPOSITORY_INFO;
-            }
-
-            // Discarded if it's not a symbolic to refs.
-            if(!head.isSymbolic()) {
-                LOGGER.fine("Unable to build RepositoryInfo. HEAD is not symbolic.");
-                LOGGER.fine("HEAD: " + head);
-                return RepositoryInfo.EMPTY_REPOSITORY_INFO;
-            }
-
-            final String defaultBranch = GitInfoUtils.normalizeBranch(head.getTarget().getName());
-            return new RepositoryInfo(defaultBranch);
+            String remoteName = getRemoteName(repository);
+            String repoUrl = getRepoUrl(repository, remoteName);
+            final String defaultBranch = getDefaultBranch(repository, remoteName);
+            String branch = GitInfoUtils.normalizeBranch(repository.getBranch()); // currently checked out branch
+            return new RepositoryInfo(repoUrl, defaultBranch, branch);
```

Comment:
If some of this attributes are `null`, I guess we're prepared for it, right?

Reply:
Yes, nothing will fail. I annotated the getters with `@Nullable`, so we'll get a warning if there's a danger of NPE.